src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -80,7 +80,7 @@ public bool IsAccessible(SmoQueryContext context)
             catch (Exception ex)
             {
                 // IsAccessible is not set of DW Gen3 so exception is expected in this case
-                if (IsDWGen3(context?.Database))
+                if (IsDWGen3(context?.Database) || IsDataMart(context?.Database))
```

Comment:
isn't this supposed to be the opposite? if DataMart then not accessible?

Reply:
isAccessible throws an exception in case of datamarts.